server.go

```diff
@@ -222,28 +222,44 @@ func (s *Server) fsScanner(done <-chan void) {
 		log.Printf("Couldn't load hashcache.dat: %v", err)
 	}
 	fastPass := (cacheSeed != nil)
-	firstPassDone := false
+
+	// Additionally autosave the hash cache each minute
+	autosaveStop := false
+	autosavePause := fastPass
+	var autosaveMtx sync.Mutex
 	go func() {
-		for range time.Tick(time.Minute) {
-			if firstPassDone {
-				return
+		// Returns true if goroutine should stop
+		autosaveFunc := func() bool {
+			autosaveMtx.Lock()
```

Comment:
It's unclear if this mutex also hold for `autosavePause`.

Reply:
Why not use RWMutex here? The only write action is at the end of the function.

Reply:
> It's unclear if this mutex also hold for `autosavePause`.

It does. Will comment accordingly.